From 82e676bb4a5e6696ba08bdec0d6d717ee2aba620 Mon Sep 17 00:00:00 2001
From: ywan171 <yi.a.wang@intel.com>
Date: Sat, 28 Jul 2012 06:08:24 +0800
Subject: RTSPSouce: add check for mHandler in release

BZ: 49222

fix the media server issue in RTSP release
duplicate quit msg may cause mHandler null and it may cause crash in onDisconnect

orig-patch:
55909(BZ:44103),RTSPSouce: add check for mHandler in release

Category: aosp improvement
Domain: Video.Media-streaming
Origin: internal
Upstream-Candidate: yes

Change-Id: Ifb324a9f9bbf443c4aa4b1ad205325478259cf4a
Orig-Change-Id: Ie00a13a5209375177ba6918f66436adf0f55e575
Signed-off-by: ywan171 <yi.a.wang@intel.com>
Signed-off-by: chuzhoux <chux.zhou@intel.com>
---
 .../libmediaplayerservice/nuplayer/RTSPSource.cpp  |    6 ++++--
 1 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/media/libmediaplayerservice/nuplayer/RTSPSource.cpp b/media/libmediaplayerservice/nuplayer/RTSPSource.cpp
index 18cf6d1..2999747 100644
--- a/media/libmediaplayerservice/nuplayer/RTSPSource.cpp
+++ b/media/libmediaplayerservice/nuplayer/RTSPSource.cpp
@@ -641,8 +641,10 @@ void NuPlayer::RTSPSource::onDisconnected(const sp<AMessage> &msg) {
     CHECK(msg->findInt32("result", &err));
     CHECK_NE(err, (status_t)OK);
 
-    mLooper->unregisterHandler(mHandler->id());
-    mHandler.clear();
+    if (mHandler != NULL) {
+        mLooper->unregisterHandler(mHandler->id());
+        mHandler.clear();
+    }
 
     if (mState == CONNECTING) {
         // We're still in the preparation phase, signal that it
-- 
1.7.4.1

